Build failure: python312Packages.patool

## Steps To Reproduce

Steps to reproduce the behavior:

1. build *python312Packages.patool*

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing pytest-check-hook
Using pytestCheckPhase
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/npfbpy7nlf8vy6s64n52vvyl3lb4s9y6-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/tests/test_version.py
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
substituteStream() in derivation python3.12-patool-2.4.0: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'patoolib/util.py')
substituteStream() in derivation python3.12-patool-2.4.0: WARNING: pattern path\ =\ None doesn't match anything in file 'patoolib/util.py'
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing setuptoolsBuildPhase
running bdist_wheel
/nix/store/sfifx7g499n2xfvs5hzvw3mxinn0i17x-python3.12-setuptools-75.3.0/lib/python3.12/site-packages/setuptools/_distutils/cmd.py:111: SetuptoolsDeprecationWarning: bdist_wheel.universal is deprecated
!!

        ********************************************************************************
        With Python 2.7 end-of-life, support for building universal wheels
        (i.e., wheels that support both Python 2 and Python 3)
        is being obviated.
        Please discontinue using this option, or if you still need it,
        file an issue with pypa/setuptools describing your use case.

        By 2025-Aug-30, you need to update your project and remove deprecated calls
        or your builds will no longer be supported.
        ********************************************************************************

!!
  self.finalize_options()
running build
running build_py
creating build/lib/patoolib
copying patoolib/__init__.py -> build/lib/patoolib
copying patoolib/cli.py -> build/lib/patoolib
copying patoolib/configuration.py -> build/lib/patoolib
copying patoolib/fileutil.py -> build/lib/patoolib
copying patoolib/log.py -> build/lib/patoolib
copying patoolib/mime.py -> build/lib/patoolib
copying patoolib/util.py -> build/lib/patoolib
creating build/lib/patoolib/programs
copying patoolib/programs/__init__.py -> build/lib/patoolib/programs
copying patoolib/programs/ar.py -> build/lib/patoolib/programs
copying patoolib/programs/arc.py -> build/lib/patoolib/programs
copying patoolib/programs/archmage.py -> build/lib/patoolib/programs
copying patoolib/programs/arj.py -> build/lib/patoolib/programs
copying patoolib/programs/bsdcpio.py -> build/lib/patoolib/programs
copying patoolib/programs/bsdtar.py -> build/lib/patoolib/programs
copying patoolib/programs/bzip2.py -> build/lib/patoolib/programs
copying patoolib/programs/bzip3.py -> build/lib/patoolib/programs
copying patoolib/programs/cabextract.py -> build/lib/patoolib/programs
copying patoolib/programs/chmlib.py -> build/lib/patoolib/programs
copying patoolib/programs/clzip.py -> build/lib/patoolib/programs
copying patoolib/programs/compress.py -> build/lib/patoolib/programs
copying patoolib/programs/cpio.py -> build/lib/patoolib/programs
copying patoolib/programs/dpkg.py -> build/lib/patoolib/programs
copying patoolib/programs/flac.py -> build/lib/patoolib/programs
copying patoolib/programs/genisoimage.py -> build/lib/patoolib/programs
copying patoolib/programs/gzip.py -> build/lib/patoolib/programs
copying patoolib/programs/isoinfo.py -> build/lib/patoolib/programs
copying patoolib/programs/jar.py -> build/lib/patoolib/programs
copying patoolib/programs/lbzip2.py -> build/lib/patoolib/programs
copying patoolib/programs/lcab.py -> build/lib/patoolib/programs
copying patoolib/programs/lha.py -> build/lib/patoolib/programs
copying patoolib/programs/lhasa.py -> build/lib/patoolib/programs
copying patoolib/programs/lrzip.py -> build/lib/patoolib/programs
copying patoolib/programs/lz4.py -> build/lib/patoolib/programs
copying patoolib/programs/lzip.py -> build/lib/patoolib/programs
copying patoolib/programs/lzma.py -> build/lib/patoolib/programs
copying patoolib/programs/lzop.py -> build/lib/patoolib/programs
copying patoolib/programs/mac.py -> build/lib/patoolib/programs
copying patoolib/programs/nomarch.py -> build/lib/patoolib/programs
copying patoolib/programs/p7azip.py -> build/lib/patoolib/programs
copying patoolib/programs/p7rzip.py -> build/lib/patoolib/programs
copying patoolib/programs/p7zip.py -> build/lib/patoolib/programs
copying patoolib/programs/p7zz.py -> build/lib/patoolib/programs
copying patoolib/programs/p7zzs.py -> build/lib/patoolib/programs
copying patoolib/programs/pbzip2.py -> build/lib/patoolib/programs
copying patoolib/programs/pdlzip.py -> build/lib/patoolib/programs
copying patoolib/programs/pigz.py -> build/lib/patoolib/programs
copying patoolib/programs/plzip.py -> build/lib/patoolib/programs
copying patoolib/programs/py_bz2.py -> build/lib/patoolib/programs
copying patoolib/programs/py_echo.py -> build/lib/patoolib/programs
copying patoolib/programs/py_gzip.py -> build/lib/patoolib/programs
copying patoolib/programs/py_lzma.py -> build/lib/patoolib/programs
copying patoolib/programs/py_tarfile.py -> build/lib/patoolib/programs
copying patoolib/programs/py_zipfile.py -> build/lib/patoolib/programs
copying patoolib/programs/rar.py -> build/lib/patoolib/programs
copying patoolib/programs/rpm.py -> build/lib/patoolib/programs
copying patoolib/programs/rpm2cpio.py -> build/lib/patoolib/programs
copying patoolib/programs/rzip.py -> build/lib/patoolib/programs
copying patoolib/programs/shar.py -> build/lib/patoolib/programs
copying patoolib/programs/shorten.py -> build/lib/patoolib/programs
copying patoolib/programs/star.py -> build/lib/patoolib/programs
copying patoolib/programs/tar.py -> build/lib/patoolib/programs
copying patoolib/programs/unace.py -> build/lib/patoolib/programs
copying patoolib/programs/unadf.py -> build/lib/patoolib/programs
copying patoolib/programs/unalz.py -> build/lib/patoolib/programs
copying patoolib/programs/unar.py -> build/lib/patoolib/programs
copying patoolib/programs/uncompress.py -> build/lib/patoolib/programs
copying patoolib/programs/unrar.py -> build/lib/patoolib/programs
copying patoolib/programs/unshar.py -> build/lib/patoolib/programs
copying patoolib/programs/unzip.py -> build/lib/patoolib/programs
copying patoolib/programs/xdms.py -> build/lib/patoolib/programs
copying patoolib/programs/xz.py -> build/lib/patoolib/programs
copying patoolib/programs/zip.py -> build/lib/patoolib/programs
copying patoolib/programs/zoo.py -> build/lib/patoolib/programs
copying patoolib/programs/zopfli.py -> build/lib/patoolib/programs
copying patoolib/programs/zpaq.py -> build/lib/patoolib/programs
copying patoolib/programs/zstd.py -> build/lib/patoolib/programs
/nix/store/sfifx7g499n2xfvs5hzvw3mxinn0i17x-python3.12-setuptools-75.3.0/lib/python3.12/site-packages/setuptools/_distutils/cmd.py:66: SetuptoolsDeprecationWarning: setup.py install is deprecated.
!!

        ********************************************************************************
        Please avoid running ``setup.py`` directly.
        Instead, use pypa/build, pypa/installer or other
        standards-based tools.

        See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
        ********************************************************************************

!!
  self.initialize_options()
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/patoolib
copying build/lib/patoolib/__init__.py -> build/bdist.linux-x86_64/wheel/./patoolib
copying build/lib/patoolib/cli.py -> build/bdist.linux-x86_64/wheel/./patoolib
copying build/lib/patoolib/configuration.py -> build/bdist.linux-x86_64/wheel/./patoolib
copying build/lib/patoolib/fileutil.py -> build/bdist.linux-x86_64/wheel/./patoolib
copying build/lib/patoolib/log.py -> build/bdist.linux-x86_64/wheel/./patoolib
copying build/lib/patoolib/mime.py -> build/bdist.linux-x86_64/wheel/./patoolib
copying build/lib/patoolib/util.py -> build/bdist.linux-x86_64/wheel/./patoolib
creating build/bdist.linux-x86_64/wheel/patoolib/programs
copying build/lib/patoolib/programs/__init__.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/ar.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/arc.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/archmage.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/arj.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/bsdcpio.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/bsdtar.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/bzip2.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/bzip3.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/cabextract.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/chmlib.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/clzip.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/compress.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/cpio.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/dpkg.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/flac.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/genisoimage.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/gzip.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/isoinfo.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/jar.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/lbzip2.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/lcab.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/lha.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/lhasa.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/lrzip.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/lz4.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/lzip.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/lzma.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/lzop.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/mac.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/nomarch.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/p7azip.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/p7rzip.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/p7zip.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/p7zz.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/p7zzs.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/pbzip2.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/pdlzip.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/pigz.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/plzip.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/py_bz2.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/py_echo.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/py_gzip.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/py_lzma.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/py_tarfile.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/py_zipfile.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/rar.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/rpm.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/rpm2cpio.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/rzip.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/shar.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/shorten.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/star.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/tar.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/unace.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/unadf.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/unalz.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/unar.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/uncompress.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/unrar.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/unshar.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/unzip.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/xdms.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/xz.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/zip.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/zoo.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/zopfli.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/zpaq.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
copying build/lib/patoolib/programs/zstd.py -> build/bdist.linux-x86_64/wheel/./patoolib/programs
running install_egg_info
running egg_info
creating patool.egg-info
writing patool.egg-info/PKG-INFO
writing dependency_links to patool.egg-info/dependency_links.txt
writing entry points to patool.egg-info/entry_points.txt
writing top-level names to patool.egg-info/top_level.txt
writing manifest file 'patool.egg-info/SOURCES.txt'
reading manifest file 'patool.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no previously-included files matching '*.py[cod]' found anywhere in distribution
adding license file 'COPYING'
writing manifest file 'patool.egg-info/SOURCES.txt'
Copying patool.egg-info to build/bdist.linux-x86_64/wheel/./patool-2.4.1-py3.12.egg-info
running install_scripts
creating build/bdist.linux-x86_64/wheel/patool-2.4.1.dist-info/WHEEL
creating 'dist/patool-2.4.1-py2.py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'patoolib/__init__.py'
adding 'patoolib/cli.py'
adding 'patoolib/configuration.py'
adding 'patoolib/fileutil.py'
adding 'patoolib/log.py'
adding 'patoolib/mime.py'
adding 'patoolib/util.py'
adding 'patoolib/programs/__init__.py'
adding 'patoolib/programs/ar.py'
adding 'patoolib/programs/arc.py'
adding 'patoolib/programs/archmage.py'
adding 'patoolib/programs/arj.py'
adding 'patoolib/programs/bsdcpio.py'
adding 'patoolib/programs/bsdtar.py'
adding 'patoolib/programs/bzip2.py'
adding 'patoolib/programs/bzip3.py'
adding 'patoolib/programs/cabextract.py'
adding 'patoolib/programs/chmlib.py'
adding 'patoolib/programs/clzip.py'
adding 'patoolib/programs/compress.py'
adding 'patoolib/programs/cpio.py'
adding 'patoolib/programs/dpkg.py'
adding 'patoolib/programs/flac.py'
adding 'patoolib/programs/genisoimage.py'
adding 'patoolib/programs/gzip.py'
adding 'patoolib/programs/isoinfo.py'
adding 'patoolib/programs/jar.py'
adding 'patoolib/programs/lbzip2.py'
adding 'patoolib/programs/lcab.py'
adding 'patoolib/programs/lha.py'
adding 'patoolib/programs/lhasa.py'
adding 'patoolib/programs/lrzip.py'
adding 'patoolib/programs/lz4.py'
adding 'patoolib/programs/lzip.py'
adding 'patoolib/programs/lzma.py'
adding 'patoolib/programs/lzop.py'
adding 'patoolib/programs/mac.py'
adding 'patoolib/programs/nomarch.py'
adding 'patoolib/programs/p7azip.py'
adding 'patoolib/programs/p7rzip.py'
adding 'patoolib/programs/p7zip.py'
adding 'patoolib/programs/p7zz.py'
adding 'patoolib/programs/p7zzs.py'
adding 'patoolib/programs/pbzip2.py'
adding 'patoolib/programs/pdlzip.py'
adding 'patoolib/programs/pigz.py'
adding 'patoolib/programs/plzip.py'
adding 'patoolib/programs/py_bz2.py'
adding 'patoolib/programs/py_echo.py'
adding 'patoolib/programs/py_gzip.py'
adding 'patoolib/programs/py_lzma.py'
adding 'patoolib/programs/py_tarfile.py'
adding 'patoolib/programs/py_zipfile.py'
adding 'patoolib/programs/rar.py'
adding 'patoolib/programs/rpm.py'
adding 'patoolib/programs/rpm2cpio.py'
adding 'patoolib/programs/rzip.py'
adding 'patoolib/programs/shar.py'
adding 'patoolib/programs/shorten.py'
adding 'patoolib/programs/star.py'
adding 'patoolib/programs/tar.py'
adding 'patoolib/programs/unace.py'
adding 'patoolib/programs/unadf.py'
adding 'patoolib/programs/unalz.py'
adding 'patoolib/programs/unar.py'
adding 'patoolib/programs/uncompress.py'
adding 'patoolib/programs/unrar.py'
adding 'patoolib/programs/unshar.py'
adding 'patoolib/programs/unzip.py'
adding 'patoolib/programs/xdms.py'
adding 'patoolib/programs/xz.py'
adding 'patoolib/programs/zip.py'
adding 'patoolib/programs/zoo.py'
adding 'patoolib/programs/zopfli.py'
adding 'patoolib/programs/zpaq.py'
adding 'patoolib/programs/zstd.py'
adding 'patool-2.4.1.dist-info/COPYING'
adding 'patool-2.4.1.dist-info/METADATA'
adding 'patool-2.4.1.dist-info/WHEEL'
adding 'patool-2.4.1.dist-info/entry_points.txt'
adding 'patool-2.4.1.dist-info/top_level.txt'
adding 'patool-2.4.1.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Finished executing setuptoolsBuildPhase
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Executing pypaInstallPhase
Successfully installed patool-2.4.1-py2.py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
@nix { "action": "setPhase", "phase": "pythonOutputDistPhase" }
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
shrinking RPATHs of ELF executables and libraries in /nix/store/1g9mj0dl9gaqx5czbwwjxymi8szim23y-python3.12-patool-2.4.0
checking for references to /build/ in /nix/store/1g9mj0dl9gaqx5czbwwjxymi8szim23y-python3.12-patool-2.4.0...
patching script interpreter paths in /nix/store/1g9mj0dl9gaqx5czbwwjxymi8szim23y-python3.12-patool-2.4.0
stripping (with command strip and flags -S -p) in  /nix/store/1g9mj0dl9gaqx5czbwwjxymi8szim23y-python3.12-patool-2.4.0/lib /nix/store/1g9mj0dl9gaqx5czbwwjxymi8szim23y-python3.12-patool-2.4.0/bin
shrinking RPATHs of ELF executables and libraries in /nix/store/h0rvydbpqpxnl46x5lqi6q9q88aykqhc-python3.12-patool-2.4.0-dist
checking for references to /build/ in /nix/store/h0rvydbpqpxnl46x5lqi6q9q88aykqhc-python3.12-patool-2.4.0-dist...
patching script interpreter paths in /nix/store/h0rvydbpqpxnl46x5lqi6q9q88aykqhc-python3.12-patool-2.4.0-dist
Rewriting #!/nix/store/c9m6yd8fg1flz2j5r4bif1ib5j20a0cy-python3-3.12.8/bin/python3.12 to #!/nix/store/c9m6yd8fg1flz2j5r4bif1ib5j20a0cy-python3-3.12.8
wrapping `/nix/store/1g9mj0dl9gaqx5czbwwjxymi8szim23y-python3.12-patool-2.4.0/bin/patool'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
@nix { "action": "setPhase", "phase": "pythonCatchConflictsPhase" }
Running phase: pythonRemoveBinBytecodePhase
@nix { "action": "setPhase", "phase": "pythonRemoveBinBytecodePhase" }
Running phase: pythonImportsCheckPhase
@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
Executing pythonImportsCheckPhase
Running phase: pytestCheckPhase
@nix { "action": "setPhase", "phase": "pytestCheckPhase" }
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.12.8, pytest-8.3.3, pluggy-1.5.0
rootdir: /build/source
configfile: pyproject.toml
collected 218 items / 6 deselected / 212 selected                              

tests/archives/test_7z.py ..ss..ss                                       [  3%]
tests/archives/test_7za.py ..                                            [  4%]
tests/archives/test_7zr.py ..                                            [  5%]
tests/archives/test_7zz.py ssssssss                                      [  9%]
tests/archives/test_7zzs.py ssssssss                                     [ 13%]
tests/archives/test_ar.py ..                                             [ 14%]
tests/archives/test_arc.py ssss                                          [ 16%]
tests/archives/test_archmage.py ss                                       [ 16%]
tests/archives/test_arj.py ssss                                          [ 18%]
tests/archives/test_bsdcpio.py ss                                        [ 19%]
tests/archives/test_bsdtar.py sssssssss                                  [ 24%]
tests/archives/test_bzip2.py ..                                          [ 25%]
tests/archives/test_bzip3.py ss                                          [ 25%]
tests/archives/test_cabextract.py ..                                     [ 26%]
tests/archives/test_chmlib.py ss                                         [ 27%]
tests/archives/test_clzip.py ss                                          [ 28%]
tests/archives/test_compress.py s                                        [ 29%]
tests/archives/test_cpio.py ss                                           [ 30%]
tests/archives/test_dpkg.py ss                                           [ 31%]
tests/archives/test_flac.py ss                                           [ 32%]
tests/archives/test_genisoimage.py s                                     [ 32%]
tests/archives/test_gzip.py ..                                           [ 33%]
tests/archives/test_isoinfo.py ss                                        [ 34%]
tests/archives/test_jar.py ss                                            [ 35%]
tests/archives/test_lbzip2.py ss                                         [ 36%]
tests/archives/test_lcab.py s                                            [ 36%]
tests/archives/test_lha.py s                                             [ 37%]
tests/archives/test_lhasa.py s                                           [ 37%]
tests/archives/test_lrzip.py s                                           [ 38%]
tests/archives/test_lz4.py ss                                            [ 39%]
tests/archives/test_lzip.py ..                                           [ 40%]
tests/archives/test_lzma.py .                                            [ 40%]
tests/archives/test_lzop.py ss                                           [ 41%]
tests/archives/test_mac.py ss                                            [ 42%]
tests/archives/test_nomarch.py ss                                        [ 43%]
tests/archives/test_pbzip2.py ss                                         [ 44%]
tests/archives/test_pdlzip.py ss                                         [ 45%]
tests/archives/test_pigz.py ss                                           [ 46%]
tests/archives/test_plzip.py ss                                          [ 47%]
tests/archives/test_pybz2.py ..                                          [ 48%]
tests/archives/test_pyecho.py ..                                         [ 49%]
tests/archives/test_pygzip.py ..                                         [ 50%]
tests/archives/test_pylzma.py ..                                         [ 50%]
tests/archives/test_pytarfile.py ........                                [ 54%]
tests/archives/test_pyzipfile.py ....                                    [ 56%]
tests/archives/test_rar.py ssss                                          [ 58%]
tests/archives/test_rpm.py ss                                            [ 59%]
tests/archives/test_rpm2cpio.py ss                                       [ 60%]
tests/archives/test_rzip.py ss                                           [ 61%]
tests/archives/test_shar.py s                                            [ 61%]
tests/archives/test_shorten.py s                                         [ 62%]
tests/archives/test_star.py sssssssssssss                                [ 68%]
tests/archives/test_tar.py ...........ss                                 [ 74%]
tests/archives/test_unace.py ss                                          [ 75%]
tests/archives/test_unadf.py ss                                          [ 76%]
tests/archives/test_unalz.py s                                           [ 76%]
tests/archives/test_unar.py ...                                          [ 78%]
tests/archives/test_uncompressreal.py ss                                 [ 79%]
tests/archives/test_unrar.py ss                                          [ 80%]
tests/archives/test_unshar.py s                                          [ 80%]
tests/archives/test_xdms.py s                                            [ 81%]
tests/archives/test_xz.py ...                                            [ 82%]
tests/archives/test_zoo.py ss                                            [ 83%]
tests/archives/test_zopfli.py s                                          [ 83%]
tests/archives/test_zpaq.py ..                                           [ 84%]
tests/test_config.py ......                                              [ 87%]
tests/test_create.py .                                                   [ 88%]
tests/test_diff.py s                                                     [ 88%]
tests/test_extract.py .                                                  [ 89%]
tests/test_fileutil.py .....                                             [ 91%]
tests/test_formats.py .                                                  [ 91%]
tests/test_help.py .                                                     [ 92%]
tests/test_list.py .                                                     [ 92%]
tests/test_mime.py ....F                                                 [ 95%]
tests/test_repack.py ...                                                 [ 96%]
tests/test_search.py .s                                                  [ 97%]
tests/test_test.py .                                                     [ 98%]
tests/test_util.py ...                                                   [ 99%]
tests/test_version.py .                                                  [100%]

=================================== FAILURES ===================================
__________________________ TestMime.test_nested_gzip ___________________________

args = (<tests.test_mime.TestMime testMethod=test_nested_gzip>,), kwargs = {}

    def newfunc(*args, **kwargs):
        if not testfunc(name):
            pytest.skip(f"{description} {name!r} is not available")
>       return func(*args, **kwargs)

tests/__init__.py:44: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_mime.py:198: in test_nested_gzip
    self.mime_test_file("t.rar.gz", "application/x-rar", "gzip")
tests/test_mime.py:56: in mime_test_file
    self.mime_test(patoolib.mime.guess_mime_file, archive, mime, encoding, info)
tests/test_mime.py:38: in mime_test
    self.assertEqual(
E   AssertionError: 'application/gzip' != 'application/x-rar'
E   - application/gzip
E   + application/x-rar
E    : MIME type for archive `/build/source/tests/data/t.rar.gz' should be application/x-rar, but was application/gzip. Info: application/x-gzip
=========================== short test summary info ============================
FAILED tests/test_mime.py::TestMime::test_nested_gzip - AssertionError: 'application/gzip' != 'application/x-rar'
=========== 1 failed, 87 passed, 124 skipped, 6 deselected in 15.99s ===========
```

</details>

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.6, NixOS, 25.05 (Warbler), 25.05.20241223.585f762`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/xvjs4p3dwwriqbyrvwvgj9zg8i8g7420-source`
```

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@marius851000 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/patool/default.nix
+++ b/pkgs/development/python-modules/patool/default.nix
@@ -38,7 +38,7 @@ let
 in
 buildPythonPackage rec {
   pname = "patool";
-  version = "2.4.0";
+  version = "3.1.0";
   format = "setuptools";
 
   #pypi doesn't have test data
@@ -46,8 +46,13 @@ buildPythonPackage rec {
     owner = "wummel";
     repo = pname;
     rev = "refs/tags/${version}";
-    hash = "sha256-dWyC8uTVUb06liNcJaG4oK9wqIDmwAl6i6cg4XRRRdQ=";
+    hash = "sha256-mt/GUIRJHB2/Rritc+uNkolZzguYy2G/NKnSKNxKsLk=";
   };
+
+  patches = [
+    # https://github.com/wummel/patool/pull/173
+    ./fix-rar-detection.patch
+  ];
 
   postPatch = ''
     substituteInPlace patoolib/util.py \
@@ -61,6 +66,10 @@ buildPythonPackage rec {
     "test_unzip_file"
     "test_zip"
     "test_zip_file"
+    "test_7z"
+    "test_7z_file"
+    "test_7za_file"
+    "test_p7azip"
   ] ++ lib.optionals stdenv.hostPlatform.isDarwin [ "test_ar" ];
 
   meta = with lib; {

--- a/pkgs/development/python-modules/patool/fix-rar-detection.patch
+++ b/pkgs/development/python-modules/patool/fix-rar-detection.patch
@@ -1,0 +1,318 @@
+From d10324eac4429ff3d7d38ad24a19210699229e07 Mon Sep 17 00:00:00 2001
+From: Alfred Wingate <parona@protonmail.com>
+Date: Wed, 11 Dec 2024 06:34:36 +0200
+Subject: [PATCH 1/4] Use application/gzip as the preferred mimetype for gzip
+
+Signed-off-by: Alfred Wingate <parona@protonmail.com>
+---
+ patoolib/__init__.py |  1 -
+ patoolib/mime.py     | 16 +++++++++++++---
+ tests/test_mime.py   |  4 ++--
+ 3 files changed, 15 insertions(+), 6 deletions(-)
+
+diff --git a/patoolib/__init__.py b/patoolib/__init__.py
+index d665a28a..2247b615 100644
+--- a/patoolib/__init__.py
++++ b/patoolib/__init__.py
+@@ -127,7 +127,6 @@
+     'application/x-cpio': 'cpio',
+     'application/x-debian-package': 'deb',
+     'application/x-dms': 'dms',
+-    'application/x-gzip': 'gzip',
+     'application/x-iso9660-image': 'iso',
+     'application/x-lz4': 'lz4',
+     'application/x-lzop': 'lzop',
+diff --git a/patoolib/mime.py b/patoolib/mime.py
+index 12405ada..c9d8894b 100644
+--- a/patoolib/mime.py
++++ b/patoolib/mime.py
+@@ -135,9 +135,11 @@ def guess_mime(filename: str) -> tuple[str | None, str | None]:
+ Mime2Encoding: dict[str, str] = dict(
+     [(_val, _key) for _key, _val in Encoding2Mime.items()]
+ )
+-# libmagic before version 5.14 identified .gz files as application/x-gzip
+-Mime2Encoding['application/x-gzip'] = 'gzip'
+ 
++LegacyMimeType: dict[str, str] = {
++    # libmagic before version 5.14 identified .gz files as application/x-gzip
++    'application/x-gzip': "application/gzip",
++}
+ 
+ def guess_mime_mimedb(filename: str) -> tuple[str | None, str | None]:
+     """Guess MIME type from given filename.
+@@ -192,6 +194,10 @@ def guess_mime_file(filename: str) -> tuple[str | None, str | None]:
+         except (OSError, subprocess.CalledProcessError) as err:
+             log_warning(f"error executing {cmd}: {err}")
+             mime2 = None
++
++        if mime2 in LegacyMimeType:
++            mime2 = LegacyMimeType[mime2]
++
+         # Some file(1) implementations return an empty or unknown mime type
+         # when the uncompressor program is not installed, other
+         # implementation return the original file type.
+@@ -227,6 +233,10 @@ def guess_mime_file_mime(
+     except OSError as err:
+         # ignore errors, as file(1) is only a fallback
+         log_warning(f"error executing {cmd}: {err}")
++
++    if mime in LegacyMimeType:
++        mime = LegacyMimeType[mime]
++
+     if mime not in ArchiveMimetypes:
+         mime, encoding = None, None
+     return mime, encoding
+@@ -253,7 +263,7 @@ def get_file_mime_encoding(parts: Sequence[str]) -> str | None:
+     "cpio archive": "application/x-cpio",
+     "ASCII cpio archive": "application/x-cpio",
+     "Debian binary package": "application/x-debian-package",
+-    "gzip compressed data": "application/x-gzip",
++    "gzip compressed data": "application/gzip",
+     "LZMA compressed data": "application/x-lzma",
+     "LRZIP compressed data": "application/x-lrzip",
+     "lzop compressed data": "application/x-lzop",
+diff --git a/tests/test_mime.py b/tests/test_mime.py
+index 40e73edf..3f292dfc 100644
+--- a/tests/test_mime.py
++++ b/tests/test_mime.py
+@@ -81,8 +81,8 @@ def test_mime_file(self):
+         self.mime_test_file("t.cpio.foo", "application/x-cpio")
+         self.mime_test_file("t.deb", "application/x-debian-package")
+         self.mime_test_file("t.deb.foo", "application/x-debian-package")
+-        self.mime_test_file("t.txt.gz", ("application/gzip", "application/x-gzip"))
+-        self.mime_test_file("t.txt.gz.foo", ("application/gzip", "application/x-gzip"))
++        self.mime_test_file("t.txt.gz", "application/gzip")
++        self.mime_test_file("t.txt.gz.foo", "application/gzip")
+         self.mime_test_file("t.jar", "application/zip")
+         self.mime_test_file("t.jar.foo", "application/zip")
+         self.mime_test_file("t.txt.lzma", "application/x-lzma")
+
+From e7501d1c7805696ff5b2ecc779f7a56ab2425c3f Mon Sep 17 00:00:00 2001
+From: Alfred Wingate <parona@protonmail.com>
+Date: Wed, 11 Dec 2024 06:37:54 +0200
+Subject: [PATCH 2/4] Change rar mime type to application/vnd.rar
+
+https://github.com/file/file/commit/d46a1f3dbbf58eb510c1779b8bdcc59d5ee24ab9
+https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1068700
+
+Signed-off-by: Alfred Wingate <parona@protonmail.com>
+---
+ patoolib/__init__.py |  2 +-
+ patoolib/mime.py     |  9 ++++++---
+ tests/test_mime.py   | 16 ++++++++--------
+ 3 files changed, 15 insertions(+), 12 deletions(-)
+
+diff --git a/patoolib/__init__.py b/patoolib/__init__.py
+index 2247b615..8b13181c 100644
+--- a/patoolib/__init__.py
++++ b/patoolib/__init__.py
+@@ -136,7 +136,6 @@
+     'application/x-lrzip': 'lrzip',
+     'application/x-lzh': 'lzh',
+     'application/x-ms-wim': 'wim',
+-    'application/x-rar': 'rar',
+     'application/x-redhat-package-manager': 'rpm',
+     'application/x-rpm': 'rpm',
+     'application/x-rzip': 'rzip',
+@@ -147,6 +146,7 @@
+     'application/x-xz': 'xz',
+     'application/x-zip-compressed': 'zip',
+     'application/x-zoo': 'zoo',
++    'application/vnd.rar': 'rar',
+     'application/zip': 'zip',
+     'application/zpaq': 'zpaq',
+     "application/zstd": "zstd",
+diff --git a/patoolib/mime.py b/patoolib/mime.py
+index c9d8894b..8de9b180 100644
+--- a/patoolib/mime.py
++++ b/patoolib/mime.py
+@@ -54,8 +54,8 @@ def add_mimedb_data(mimedb: mimetypes.MimeTypes) -> None:
+     add_mimetype(mimedb, 'application/x-lzma', '.lzma')
+     add_mimetype(mimedb, 'application/x-xz', '.xz')
+     add_mimetype(mimedb, 'application/java-archive', '.jar')
+-    add_mimetype(mimedb, 'application/x-rar', '.rar')
+-    add_mimetype(mimedb, 'application/x-rar', '.cbr')
++    add_mimetype(mimedb, 'application/vnd.rar', '.rar')
++    add_mimetype(mimedb, 'application/vnd.rar', '.cbr')
+     add_mimetype(mimedb, 'application/x-7z-compressed', '.7z')
+     add_mimetype(mimedb, 'application/x-7z-compressed', '.cb7')
+     add_mimetype(mimedb, 'application/x-cab', '.cab')
+@@ -139,8 +139,11 @@ def guess_mime(filename: str) -> tuple[str | None, str | None]:
+ LegacyMimeType: dict[str, str] = {
+     # libmagic before version 5.14 identified .gz files as application/x-gzip
+     'application/x-gzip': "application/gzip",
++    # libmagic before version 5.46 identified .rar files as application/x-rar
++    'application/x-rar': "application/vnd.rar",
+ }
+ 
++
+ def guess_mime_mimedb(filename: str) -> tuple[str | None, str | None]:
+     """Guess MIME type from given filename.
+     @return: tuple (mime, encoding)
+@@ -268,7 +271,7 @@ def get_file_mime_encoding(parts: Sequence[str]) -> str | None:
+     "LRZIP compressed data": "application/x-lrzip",
+     "lzop compressed data": "application/x-lzop",
+     "Microsoft Cabinet archive data": "application/vnd.ms-cab-compressed",
+-    "RAR archive data": "application/x-rar",
++    "RAR archive data": "application/vnd.rar",
+     "RPM ": "application/x-redhat-package-manager",
+     "POSIX tar archive": "application/x-tar",
+     "xz compressed data": "application/x-xz",
+diff --git a/tests/test_mime.py b/tests/test_mime.py
+index 3f292dfc..8f3ecb8c 100644
+--- a/tests/test_mime.py
++++ b/tests/test_mime.py
+@@ -91,10 +91,10 @@ def test_mime_file(self):
+         self.mime_test_file("t.txt.lz.foo", "application/x-lzip")
+         self.mime_test_file("t.txt.lzo", "application/x-lzop")
+         self.mime_test_file("t.txt.lzo.foo", "application/x-lzop")
+-        self.mime_test_file("t.rar", "application/x-rar")
+-        self.mime_test_file("t.rar.foo", "application/x-rar")
+-        self.mime_test_file("t.cbr", "application/x-rar")
+-        self.mime_test_file("t.cbr.foo", "application/x-rar")
++        self.mime_test_file("t.rar", "application/vnd.rar")
++        self.mime_test_file("t.rar.foo", "application/vnd.rar")
++        self.mime_test_file("t.cbr", "application/vnd.rar")
++        self.mime_test_file("t.cbr.foo", "application/vnd.rar")
+         self.mime_test_file("t.rpm", "application/x-rpm")
+         self.mime_test_file("t.rpm.foo", "application/x-rpm")
+         self.mime_test_file("t.tar", "application/x-tar")
+@@ -197,8 +197,8 @@ def test_nested_gzip(self):
+         """Test mime detection of archives with double compression"""
+         # We won't extract this with rar, as it doesn't support archives wrapped in gzip
+         # compression, but we will recognize the archive as a gzip-wrapped rar-file
+-        self.mime_test_file("t.rar.gz", "application/x-rar", "gzip")
+-        self.mime_test_file("t.rar.gz.foo", "application/x-rar", "gzip")
++        self.mime_test_file("t.rar.gz", "application/vnd.rar", "gzip")
++        self.mime_test_file("t.rar.gz.foo", "application/vnd.rar", "gzip")
+ 
+     @needs_program('file')
+     @needs_program('gzip')
+@@ -237,7 +237,7 @@ def test_mime_mimedb(self):
+         self.mime_test_mimedb("t .bz2", "application/x-bzip2")
+         self.mime_test_mimedb("t .bz3", "application/x-bzip3")
+         self.mime_test_mimedb("t.cab", "application/x-cab")
+-        self.mime_test_mimedb("t.cbr", ("application/rar", "application/x-rar"))
++        self.mime_test_mimedb("t.cbr", ("application/rar", "application/vnd.rar"))
+         self.mime_test_mimedb("t.cpio", "application/x-cpio")
+         self.mime_test_mimedb("t.deb", "application/x-debian-package")
+         self.mime_test_mimedb("t.gz", "application/gzip")
+@@ -247,7 +247,7 @@ def test_mime_mimedb(self):
+         self.mime_test_mimedb("t.txt.lz", "application/x-lzip")
+         self.mime_test_mimedb("t.txt.lz4", "application/x-lz4")
+         self.mime_test_mimedb("t.lzo", "application/x-lzop")
+-        self.mime_test_mimedb("t.rar", ("application/rar", "application/x-rar"))
++        self.mime_test_mimedb("t.rar", ("application/rar", "application/vnd.rar"))
+         self.mime_test_mimedb(
+             "t.rpm", ("application/x-redhat-package-manager", "application/x-rpm")
+         )
+
+From 85fafd16ec01a7eb793e04011617bb47211d446a Mon Sep 17 00:00:00 2001
+From: Alfred Wingate <parona@protonmail.com>
+Date: Wed, 11 Dec 2024 06:40:41 +0200
+Subject: [PATCH 3/4] Remove references to application/rar
+
+application/vnd.rar is the IANA assigned mime type for rar. libmagic has
+never refered to application/rar either, so there is no use for backwards
+compatibility.
+
+https://www.iana.org/assignments/media-types/application/vnd.rar
+
+Signed-off-by: Alfred Wingate <parona@protonmail.com>
+---
+ patoolib/__init__.py | 1 -
+ tests/test_mime.py   | 4 ++--
+ 2 files changed, 2 insertions(+), 3 deletions(-)
+
+diff --git a/patoolib/__init__.py b/patoolib/__init__.py
+index 8b13181c..51f8b4eb 100644
+--- a/patoolib/__init__.py
++++ b/patoolib/__init__.py
+@@ -110,7 +110,6 @@
+     'application/jar': 'zip',  # reported on older systems such as ubuntu 14.04
+     'application/java-archive': 'zip',
+     'application/vnd.android.package-archive': 'zip',
+-    'application/rar': 'rar',
+     'application/vnd.ms-cab-compressed': 'cab',
+     'application/x-7z-compressed': '7z',
+     'application/x-ace': 'ace',
+diff --git a/tests/test_mime.py b/tests/test_mime.py
+index 8f3ecb8c..287d8208 100644
+--- a/tests/test_mime.py
++++ b/tests/test_mime.py
+@@ -237,7 +237,7 @@ def test_mime_mimedb(self):
+         self.mime_test_mimedb("t .bz2", "application/x-bzip2")
+         self.mime_test_mimedb("t .bz3", "application/x-bzip3")
+         self.mime_test_mimedb("t.cab", "application/x-cab")
+-        self.mime_test_mimedb("t.cbr", ("application/rar", "application/vnd.rar"))
++        self.mime_test_mimedb("t.cbr", "application/vnd.rar")
+         self.mime_test_mimedb("t.cpio", "application/x-cpio")
+         self.mime_test_mimedb("t.deb", "application/x-debian-package")
+         self.mime_test_mimedb("t.gz", "application/gzip")
+@@ -247,7 +247,7 @@ def test_mime_mimedb(self):
+         self.mime_test_mimedb("t.txt.lz", "application/x-lzip")
+         self.mime_test_mimedb("t.txt.lz4", "application/x-lz4")
+         self.mime_test_mimedb("t.lzo", "application/x-lzop")
+-        self.mime_test_mimedb("t.rar", ("application/rar", "application/vnd.rar"))
++        self.mime_test_mimedb("t.rar", "application/vnd.rar")
+         self.mime_test_mimedb(
+             "t.rpm", ("application/x-redhat-package-manager", "application/x-rpm")
+         )
+
+From bab80a04a72941b3c53e5fce1f96f5a2fc531280 Mon Sep 17 00:00:00 2001
+From: Alfred Wingate <parona@protonmail.com>
+Date: Wed, 11 Dec 2024 07:04:19 +0200
+Subject: [PATCH 4/4] Remove references to application/x-redhat-package-manager
+
+libmagic has never referred to x-redhat-package-manager and has had
+x-rpm since 3.30. Red Hat themselves use x-rpm in their mailcap.
+
+https://pagure.io/mailcap/blob/master/f/mime.types
+
+Signed-off-by: Alfred Wingate <parona@protonmail.com>
+---
+ patoolib/__init__.py | 1 -
+ patoolib/mime.py     | 2 +-
+ tests/test_mime.py   | 4 +---
+ 3 files changed, 2 insertions(+), 5 deletions(-)
+
+diff --git a/patoolib/__init__.py b/patoolib/__init__.py
+index 51f8b4eb..9d8bc849 100644
+--- a/patoolib/__init__.py
++++ b/patoolib/__init__.py
+@@ -135,7 +135,6 @@
+     'application/x-lrzip': 'lrzip',
+     'application/x-lzh': 'lzh',
+     'application/x-ms-wim': 'wim',
+-    'application/x-redhat-package-manager': 'rpm',
+     'application/x-rpm': 'rpm',
+     'application/x-rzip': 'rzip',
+     'application/x-shar': 'shar',
+diff --git a/patoolib/mime.py b/patoolib/mime.py
+index 8de9b180..ffad9cd0 100644
+--- a/patoolib/mime.py
++++ b/patoolib/mime.py
+@@ -272,7 +272,7 @@ def get_file_mime_encoding(parts: Sequence[str]) -> str | None:
+     "lzop compressed data": "application/x-lzop",
+     "Microsoft Cabinet archive data": "application/vnd.ms-cab-compressed",
+     "RAR archive data": "application/vnd.rar",
+-    "RPM ": "application/x-redhat-package-manager",
++    "RPM ": "application/x-rpm",
+     "POSIX tar archive": "application/x-tar",
+     "xz compressed data": "application/x-xz",
+     "Zip archive data": "application/zip",
+diff --git a/tests/test_mime.py b/tests/test_mime.py
+index 287d8208..a337f87a 100644
+--- a/tests/test_mime.py
++++ b/tests/test_mime.py
+@@ -248,9 +248,7 @@ def test_mime_mimedb(self):
+         self.mime_test_mimedb("t.txt.lz4", "application/x-lz4")
+         self.mime_test_mimedb("t.lzo", "application/x-lzop")
+         self.mime_test_mimedb("t.rar", "application/vnd.rar")
+-        self.mime_test_mimedb(
+-            "t.rpm", ("application/x-redhat-package-manager", "application/x-rpm")
+-        )
++        self.mime_test_mimedb("t.rpm", "application/x-rpm")
+         self.mime_test_mimedb("t.tar", "application/x-tar")
+         self.mime_test_mimedb("t.cbt", "application/x-tar")
+         self.mime_test_mimedb("t.tar.bz2", "application/x-tar", "bzip2")

